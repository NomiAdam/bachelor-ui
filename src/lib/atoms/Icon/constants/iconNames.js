import {
    IoAndroidAttach,
    IoIosCogOutline,
    IoIosAlbumsOutline,
    IoIosStar,
    IoIosStarOutline,
    IoAndroidAdd,
    IoIosUndoOutline,
    IoIosPhotosOutline,
    IoIosArrowBack,
    IoAperture,
    IoIosCloseOutline,
    IoAndroidSend,
    IoIosDownloadOutline,
    IoIosTrashOutline,
    IoEye,
    IoEyeDisabled,
    IoIosSearch,
    IoIosArrowDown,
} from 'react-icons/lib/io';
import { keys } from 'ramda';

const IconTypes = {
    cog: IoIosCogOutline,
    starFill: IoIosStar,
    star: IoIosStarOutline,
    documents: IoIosAlbumsOutline,
    attach: IoAndroidAttach,
    reply: IoIosUndoOutline,
    download: IoIosDownloadOutline,
    compose: IoIosPhotosOutline,
    aperture: IoAperture,
    arrowBack: IoIosArrowBack,
    arrowDown: IoIosArrowDown,
    add: IoAndroidAdd,
    close: IoIosCloseOutline,
    send: IoAndroidSend,
    trash: IoIosTrashOutline,
    eyeOff: IoEyeDisabled,
    eye: IoEye,
    search: IoIosSearch,
};

export const iconStringTypes = keys(IconTypes);

export default IconTypes;
