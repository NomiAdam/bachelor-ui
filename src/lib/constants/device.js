const size = {
    mobileS: '320px',
    mobileM: '375px',
    mobileL: '425px',
    tablet: '768px',
    laptop: '1024px',
    laptopL: '1440px',
    desktop: '2560px'
};

export const maxMobileS = `(max-width: ${size.mobileS})`;
export const maxMobileM = `(max-width: ${size.mobileM})`;

export const minTablet = `(min-width: ${size.tablet})`;
export const maxTablet = `(max-width: ${size.tablet})`;

export default {
    mobileS: `(min-width: ${size.mobileS})`,
    mobileM: `(min-width: ${size.mobileM})`,
    mobileL: `(min-width: ${size.mobileL})`,
    tablet: `(min-width: ${size.tablet})`,
    laptop: `(min-width: ${size.laptop})`,
    laptopL: `(min-width: ${size.laptopL})`,
    desktop: `(min-width: ${size.desktop})`,
    desktopL: `(min-width: ${size.desktop})`
};