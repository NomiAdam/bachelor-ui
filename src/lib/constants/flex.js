export default {
    FLEX: 'flex',
    DIRECTION_ROW: 'row',
    DIRECTION_COLUMN: 'column',
    POSITION_LEFT: 'flex-start',
    POSITION_RIGHT: 'flex-end',
    POSITION_CENTER: 'center',
    POSITION_SPACER: 'space-between',
    POSITION_AROUND: 'space-around',
};
