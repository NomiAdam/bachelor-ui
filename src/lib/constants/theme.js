export const basicTheme = {
    blue: '#0247fe',
    red: '#ff230f',
    green: '#009933',
    purple: '#9a0794',
    orange: '#ff9933',
    grey: '#94989f',
};

export const lightTheme = {
    lightGrey: '#f5f5f5',
    white: '#ffffff',
    whiteBlue: '#c2eeff',
    lightBlue: '#0099ff',
    lightPurple: '#4424d6',
    lightGreen: '#00ad3e',
    lightOrange: '#ffbf00',
};

export const darkTheme = {
    darkBlue: '#110934',
    darkPurple: '#521751',
    darkGrey: '#1d1f1f',
    black: '#000000',
    darkOrange: '#ff4000'
};
