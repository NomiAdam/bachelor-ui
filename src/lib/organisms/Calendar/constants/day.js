export default [
    {
        items: [{
            id: 1,
            label: 'RoleOne',
            start: '6:00',
            end: '8:00',
        }],
    },
    {
        items: [],
    },
    {
        items: [{
            id: 2,
            label: 'RoleTwo',
            start: '7:00',
            end: '10:00',
        }],
    },
    {
        items: [],
    }
];

// TODO: Compute start and span a dates but calculated it differently
