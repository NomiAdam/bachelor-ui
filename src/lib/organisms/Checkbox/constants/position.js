import { keyMirror } from 'ramda-extension';
import { equals } from 'ramda';

export const LABEL_POSITION = keyMirror({
    LEFT: null,
    RIGHT: null,
});

export const equalsLeft = equals(LABEL_POSITION.LEFT);
export const equalsRight = equals(LABEL_POSITION.RIGHT);
