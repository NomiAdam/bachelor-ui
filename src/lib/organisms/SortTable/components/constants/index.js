// eslint-disable-next-line import/prefer-default-export
export const basicRowCount = [
    {
        value: 10,
        label: 10,
    },
    {
        value: 25,
        label: 25,
    },
    {
        value: 50,
        label: 50,
    },
];
