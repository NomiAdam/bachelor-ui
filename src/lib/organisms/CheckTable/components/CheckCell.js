import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { lightTheme, basicTheme } from '../../../constants/theme';

const StyledTableCell = styled.td`
    margin: 10px 0;
	background-color: ${({checked, disabled}) => disabled ? basicTheme.grey : checked ? lightTheme.lightBlue : 'transparent'};
`;

class CheckCell extends PureComponent {
    render() {
        const { checked } = this.props;
        return(
            <StyledTableCell checked={checked} />
        )
    }
}

CheckCell.propTypes = {
    /**
     * Boolean whether cell is checked or not
     */
    checked: PropTypes.bool,
    /**
     * Boolean whether cell is disabled or not
     */
    disabled: PropTypes.bool,
};

CheckCell.defaultProps = {
    checked: false,
};

export { StyledTableCell };
export default CheckCell;
