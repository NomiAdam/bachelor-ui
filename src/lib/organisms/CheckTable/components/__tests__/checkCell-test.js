/* eslint-disable no-undef */
import React from 'react';
import {shallow} from 'enzyme';
import CheckCell, { StyledTableCell } from '../CheckCell';
import 'jest-styled-components';
import { lightTheme } from '../../../../constants/theme';

describe('CheckCell', () => {
    it('should be defined', () => {
        expect(shallow(<CheckCell/>)).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<CheckCell/>)).toMatchSnapshot();
    });
    it('should be of type StyledCheckCell', () => {
        expect(shallow(<CheckCell />).type()).toBe(StyledTableCell);
    });
    it('should have transparent color on default', () => {
        const wrapper = shallow(<CheckCell />);
        expect(wrapper.find(StyledTableCell)).toHaveStyleRule('background-color', 'transparent');
    });
    it('should have blue color when checked', () => {
        const wrapper = shallow(<CheckCell checked />);
        expect(wrapper.find(StyledTableCell)).toHaveStyleRule('background-color', lightTheme.lightBlue);
    });
});
