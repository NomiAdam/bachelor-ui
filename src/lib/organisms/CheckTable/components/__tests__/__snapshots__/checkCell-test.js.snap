// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckCell should match the snapshot 1`] = `
<CheckCell__StyledTableCell
  checked={false}
/>
`;
